### PR TITLE
Implemented Find By Tags endpoint.

### LINKED ISSUE #13: Add Find By Tags Endpoint
To achieve compatibility with https://github.com/ariesprayoga/swagger-java-sample-app/blob/master/src/main/java/com/wordnik/swagger/sample/resource/PetResource.java

### PR DESCRIPTION
**Pull request by michaeltrulyng**
This pull request attempts to achieve the following:
- [x] Implement endpoint which allows multiple optional parameters for pet tag filtering (to resolve #13)
- [x] Correct swapping of `Pet: tags` and `photoUrls` parameters

The unit tests haven't been updated to reflect this new endpoint functionality.

### CODE REVIEW

File: src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala [hunk 1]
```diff
@@ -76,12 +76,26 @@ class DoobiePetRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
       .list
       .transact(xa)
 
-  override def findByStatus(statuses: NonEmptyList[PetStatus]): F[List[Pet]] = {
-    val q = sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
+  def findByStatus(statuses: NonEmptyList[PetStatus]): F[List[Pet]] =
+    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
```
Comment (pauljamescleary) [inline src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala]: do we need the parens?

File: src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala [hunk 2]
```diff
@@ -76,12 +76,26 @@ class DoobiePetRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
...
+      .transact(xa)
+
+  def findByTag(tags: NonEmptyList[String]): F[List[Pet]] = {
+    /* Handle dynamic construction of query based on multiple parameters */
 
-    q.query[Pet].list.transact(xa)
+    /* To piggyback off of comment of above reference about tags implementation, findByTag uses LIKE for partial matching
+    since tags is (currently) implemented as a comma-delimited string */
+    val tagLikeString: String = tags.toList mkString ("TAGS LIKE '%","%' OR TAGS LIKE '%","%'")
+    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
```
Comment (pauljamescleary) [inline src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala]: do we need the parens?

File: src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala (hunk 1 above)
Comment (michaeltrulyng) [reply src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala]: The parens are needed to join the SQL interpolator with the fragment.
```
    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
            FROM PET
           WHERE """ ++ Fragments.in(fr"STATUS", statuses))
      .query[Pet]
      .list
      .transact(xa)
```

File: src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala (hunk 2 above)
Comment (michaeltrulyng) [reply src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala]: The parens are needed to join the SQL Interpolator with the fragment.

```
    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
         FROM PET
         WHERE """ ++ Fragment.const(tagLikeString))
      .query[Pet]
      .list
      .transact(xa)
```

**Comment by michaeltrulyng** [comment]
A functional test has been added for the FindByTags endpoint (amended to the previous commit).

**Pull request merged**